virttest/qemu_vm.py

```diff
@@ -2140,9 +2122,18 @@ def add_disable_modern(devices, dev, dev_type):
                               "virtio-serial-pci", "virtio-rng-pci"]
         for device in devices:
             dev_type = device.get_param("driver")
-            if dev_type in virtio_pci_devices:
-                add_disable_legacy(devices, device, dev_type)
-                add_disable_modern(devices, device, dev_type)
+            options = devices.execute_qemu("-device %s,?" % dev_type)
+            host_kernel_ver = params.get("host_kernel_ver_str")
+            # Currently only need to add the options for RHEL.7.2 host, you can add them to other hosts if necessary
+            if host_kernel_ver == "Host_RHEL.7.2" and ("disable-legacy" and "disable-modern") in options:
+                set_disable_legacy_modern = True
+                disable_legacy = params.get("disable_legacy", "off")
+                disable_modern = params.get("disable_modern", "on")
+            else:
+                set_disable_legacy_modern = False
+            if set_disable_legacy_modern and dev_type in virtio_pci_devices:
+                device.set_param("disable-legacy", disable_legacy)
+                device.set_param("disable-modern", disable_modern)
 
```

Comment:
Frankly speaking, I don't like this change.
2125 ~ 2136 looks messy.
And we can declare the variable disable_legacy and disable_modern in cfg to achieve your purpose , right？